# Ville de Québec, Wikidata et livres: une exploration

## À propos

Ce carnet de note s'inscrit dans les recherches du projet [TAMIS](https://tamis.ca/) pour explorer la présence littéraire de la ville de Québec dans Wikidata. Plus de détails au sujet de cette démarche seront publiées sur le carnet de TAMIS.

## Notes techniques

### Qu'est-ce que c'est?

Ce document est un carnet Jupyter Notebook configuré pour faire des requêtes SPARQL en s'inspirant directement des [explications de Bob DuCharme](http://www.bobdc.com/blog/jupytersparql/). Les requêtes sont réalisées directement sur le «endpoint» SPAQL de Wikidata:

In [2]:
%endpoint https://query.wikidata.org/bigdata/namespace/wdq/sparql
%format any
%display table

Endpoint set to: https://query.wikidata.org/bigdata/namespace/wdq/sparql
Request format: ANY
Display: table

### Références

On s'attend à ce que les livres en général soient présents dans Wikidata en respectant les recommandations du projet livres ([anglais](https://www.wikidata.org/wiki/Wikidata:WikiProject_Books), [français](https://www.wikidata.org/wiki/Wikidata:WikiProject_Books/fr)).

Je note au passage quelques numéros d'entités qui seront utilisées fréquemment dans les requêtes qui suivront.

* Livre: Q571
* Version, édition ou traduction: Q3331189
* Maison d'éditon: Q2085381
* Oeuvre écrite (Q47461344) [note: selon le projet Wikidata Livres, les oeuvres devraient être de ce type, ou d'une de ses sous-classes, comme oeuvre littéraire Q7725634]
* Écrivain: Q36180
* Auteur: Q482980
* Poète: Q49757
* Ville de Québec: Q2145
* Quelques villes références: Montréal Q340; Paris Q90; New-York City Q60
* Quelques villes créatives de l'UNESCO en littérature: Angoulème Q134346, Barcelone Q1492, Dublin Q1761, Milan Q490, Seattle Q5083

Je note également quelques propriétés utiles.

* lieu (P276) (endroit où se situe l'objet, la structure ou l'événement. Dans le cas d'une entité administrative utiliser P131. Dans le cas d'une entité géographique utiliser P706.)
* localisation administrative (P131)
* localisation géographique (P706)
* sous-classe de (P279)
* nature de l'élément (P31)
* siège (P159)
* éditeur (P123)
* scolarité (P69)
* lieu de naissance (P19)

## Exploration sur les maisons d'édition

### De Québec

On veut avoir la liste des maisons d'édition de Québec présentes sur Wikidata (à noter l'utilisation de P159 et de P131... P159 devrait suffire, mais on trouve plus de résultats en utilisant aussi P131).

In [2]:
SELECT distinct ?publisher ?publisherLabel
WHERE
{
    ?publisher wdt:P31/wdt:P279* wd:Q2085381 .
    ?publisher (wdt:P159|wdt:P131)/wdt:P131* wd:Q2145 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}

publisher 
 publisherLabel http://www.wikidata.org/entity/Q2338135 
 Musée national des beaux-arts du Québec http://www.wikidata.org/entity/Q1705530 
 Musée de la civilisation http://www.wikidata.org/entity/Q3402572 
 Presses de l'Université Laval http://www.wikidata.org/entity/Q16506987 
 Bibliothèque de l'Université Laval http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3402573 
 Presses de l'Université du Québec http://www.wikidata.org/entity/Q95623497 
 Librairie Garneau http://www.wikidata.org/entity/Q2986483 
 Commission de la capitale nationale du Québec http://www.wikidata.org/entity/Q95690756 
 L’Œil de Poisson http://www.wikidata.org/entity/Q16684371 
 Éditions Alto http://www.wikidata.org/entity/Q3579664 
 Éditions du Septentrion http://www.wikidata.org/entity/Q3579394 
 Éditions Le Dauphin Blanc http://www.wikidata.org/entity/Q326282 
 Université du Québec http://www.wikidata.org/entity/Q40882270 
 Dussault & Proulx http://www.wikidata.org/entity/Q3479007 
 Septembre éditeur http://www.wikidata.org/entity/Q95690827 
 Le Lieu http://www.wikidata.org/entity/Q95690585 
 La Chambre blanche http://www.wikidata.org/entity/Q97044787 
 The Chronicle-Telegraph Pub. Co. Total: 18, Shown: 18

Je note la présence d'organisations disparues (Dussault & Proulx), d'organisations qu'on n'aurait pas identifiées comme maisons d'éditions _a priori_ (le Musée de la civilisation, la CCNQ ou des centres d'artistes). On voit que la fiche des Éditions Alto a été complétée depuis les observations précédentes.

Si on s'intéresse aux éditions publiées par des entités de Québec (ici, on ne spécifie pas qu'on cherche des maisons d'édition, parce qu'on part des éditions, donc on assume qu'elles sont publiées par des maisons):

In [3]:
SELECT distinct ?edition ?editionLabel ?publisher ?publisherLabel
WHERE
{
    ?edition wdt:P31/wdt:P279* wd:Q3331189 .
    ?edition wdt:P123 ?publisher .
    ?publisher (wdt:P159|wdt:P131)/wdt:P131 wd:Q2145 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}

edition 
 editionLabel 
 publisher 
 publisherLabel http://www.wikidata.org/entity/Q52182553 
 Le Journalisme à l'heure du Net 
 http://www.wikidata.org/entity/Q3402572 
 Presses de l'Université Laval http://www.wikidata.org/entity/Q58341258 
 Statistiques sociales 
 http://www.wikidata.org/entity/Q3402572 
 Presses de l'Université Laval http://www.wikidata.org/entity/Q63373321 
 Bibliothèque de l'Université Laval : 165 ans d'histoire : 1852-2017 
 http://www.wikidata.org/entity/Q3402572 
 Presses de l'Université Laval http://www.wikidata.org/entity/Q63615887 
 Dialogue avec l'histoire : les collections de la Bibliothèque de l'Université Laval 
 http://www.wikidata.org/entity/Q3402572 
 Presses de l'Université Laval http://www.wikidata.org/entity/Q77009625 
 Dictionnaire des artistes de langue française en Amérique du Nord 
 http://www.wikidata.org/entity/Q3402572 
 Presses de l'Université Laval http://www.wikidata.org/entity/Q77009625 
 Dictionnaire des artistes de langue française en Amérique du Nord 
 http://www.wikidata.org/entity/Q2338135 
 Musée national des beaux-arts du Québec Total: 6, Shown: 6

Et si on regarde les oeuvres qui sont publiées à Québec (ce qui ne serait pas une forme correcte selon le Wikiprojet Livres, mais on peut imaginer qu'il y a des entrées qui ne s'y conforment pas):

In [14]:
SELECT ?work ?workLabel
WHERE
{
    ?work wdt:P31/wdt:P279 wd:Q47461344 .
    ?publisher (wdt:P159|wdt:P131)/wdt:P131* wd:Q2145 .
    ?work wdt:P123 ?publisher .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}


Error: SPARQL error: EndPointInternalError: endpoint returned code 500 and response. 

Response:
b'SPARQL-QUERY: queryStr=SELECT ?work ?workLabel\nWHERE\n{\n    ?work wdt:P31/wdt:P279 wd:Q47461344 .\n    ?publisher (wdt:P159|wdt:P131)/wdt:P131* wd:Q2145 .\n    ?work wdt:P123 ?publisher .\n    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }\n}\n\njava.util.concurrent.TimeoutException\n\tat java.util.concurrent.FutureTask.get(FutureTask.java:205)\n\tat com.bigdata.rdf.sail.webapp.BigdataServlet.submitApiTask(BigdataServlet.java:292)\n\tat com.bigdata.rdf.sail.webapp.QueryServlet.doSparqlQuery(QueryServlet.java:678)\n\tat com.bigdata.rdf.sail.webapp.QueryServlet.doGet(QueryServlet.java:290)\n\tat com.bigdata.rdf.sail.webapp.RESTServlet.doGet(RESTServlet.java:240)\n\tat com.bigdata.rdf.sail.webapp.MultiTenancyServlet.doGet(MultiTenancyServlet.java:273)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:687)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:790)\n\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)\n\tat org.wikidata.query.rdf.blazegraph.throttling.ThrottlingFilter.doFilter(ThrottlingFilter.java:320)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.throttling.SystemOverloadFilter.doFilter(SystemOverloadFilter.java:82)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat ch.qos.logback.classic.helpers.MDCInsertingServletFilter.doFilter(MDCInsertingServletFilter.java:49)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.QueryEventSenderFilter.doFilter(QueryEventSenderFilter.java:117)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.ClientIPFilter.doFilter(ClientIPFilter.java:43)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.RealAgentFilter.doFilter(RealAgentFilter.java:33)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634)\n\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)\n\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)\n\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)\n\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)\n\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1242)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)\n\tat org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)\n\tat org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\n\tat org.eclipse.jetty.server.Server.handle(Server.java:503)\n\tat org.eclipse.jetty.server.HttpChannel.handle(HttpCh

Il faut noter que la requête n'est pas parfaite: on aurait dû mettre P279\* dans la première ligne de la clause WHERE pour identifier tous les sous-types d'oeuvre littéraire, mais ce n'était pas possible parce que ça fait en sorte que la requête n'est pas exécutée («time out»). C'est d'ailleurs le cas si on l'exécute dans ce notebook Jupyter. Dans le query service de Wikidata, on a 94 résultats.

### D'autres villes

Pour se donner des points de repère, on veut comparer avec les autres villes créatives l'UNESCO en littérature, et nos villes de référence.

In [6]:
SELECT distinct ?publisher ?publisherLabel ?city ?cityLabel
WHERE
{
    VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
    ?publisher wdt:P31/wdt:P279* wd:Q2085381 .
    ?publisher (wdt:P159|wdt:P131)/wdt:P131* ?city .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}

publisher 
 publisherLabel 
 city 
 cityLabel http://www.wikidata.org/entity/Q3976712 
 Sugar Music 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19752886 
 Q19752886 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19752958 
 Q19752958 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753302 
 Q19753302 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753311 
 Q19753311 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753313 
 Q19753313 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753320 
 Q19753320 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753419 
 Q19753419 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753417 
 Q19753417 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753423 
 Q19753423 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753437 
 Q19753437 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753458 
 Q19753458 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753468 
 Q19753468 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753474 
 Q19753474 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753472 
 Q19753472 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753484 
 Q19753484 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753492 
 Q19753492 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753498 
 Q19753498 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753503 
 Q19753503 
 http://www.wikidata.org/entity/Q490 
 Milan http://www.wikidata.org/entity/Q19753509 
 Q19753509 
 http://www.wikidata.org/entity/Q490 
 Milan Total: 1191, Shown: 20

Avec 1191 maisons d'édition dans les 9 villes échantillonnées, on a une moyenne de 132 maisons par ville (7 fois plus qu'à Québec). Si on veut faire le décompte par ville:

In [7]:
SELECT ?cityLabel (count(distinct ?publisher) as ?pubCount)
WHERE
{
    VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
    ?publisher wdt:P31/wdt:P279* wd:Q2085381 .
    ?publisher (wdt:P159|wdt:P131)/wdt:P131* ?city .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}
GROUP BY ?cityLabel

cityLabel 
 pubCount Montréal 
 61 Milan 
 165 Barcelone 
 153 Paris 
 421 New York 
 324 Québec 
 18 Dublin 
 19 Seattle 
 28 Angoulême 
 2 Total: 9, Shown: 9

Évidemment, ces données reflètent _les maisons d'édition dont la fiche Wikidata mentionne la ville_, dans la propriété P159 ou P131.

## Exploration sur les auteurs

### De Québec

Les exemples de Wikidata suggèrent de chercher des instances d'écrivain, d'auteurs ou de poètes pour trouver des auteurs de façon générale. Je tente d'abord de trouver des auteurs qui ont une relation _directe_ (sans passer par d'autres entités), peu importe le type, avec Québec:

In [11]:
SELECT distinct ?writer ?writerLabel ?relation ?relationLabel
WHERE
{
  VALUES ?occupation { wd:Q36180 wd:Q482980 wd:Q49757 }
  ?writer ?rel wd:Q2145 .
  ?writer wdt:P106 ?occupation .
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
  ?relation wikibase:directClaim ?rel .
}

writer 
 writerLabel 
 relation 
 relationLabel http://www.wikidata.org/entity/Q3059521 
 Eudore Évanturel 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q3056973 
 Ernest Gagnon 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3455481 
 Réginald Martel 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3383315 
 Pierre-Luc Lafrance 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q281608 
 Jean de Quen 
 http://www.wikidata.org/entity/P937 
 lieu de travail http://www.wikidata.org/entity/Q3336057 
 Narcisse-Henri-Édouard Faucher de Saint-Maurice 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q2958277 
 Charles Baillairgé 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3083652 
 François-Réal Angers 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3387193 
 Pierre Turgeon 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q4714622 
 Alejandro Núñez Alonso 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3320920 
 Monique Proulx 
 http://www.wikidata.org/entity/P551 
 résidence http://www.wikidata.org/entity/Q939745 
 Norah McClintock 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q2836600 
 Alice Lemieux-Lévesque 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q2977221 
 Claude Cossette 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q3438751 
 Roger Brien 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3506020 
 Suzanne Paradis 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q3439200 
 Roger Lemelin 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3439937 
 Roland Arpin 
 http://www.wikidata.org/entity/P20 
 lieu de mort http://www.wikidata.org/entity/Q3009741 
 Cécile Gagnon 
 http://www.wikidata.org/entity/P19 
 lieu de naissance http://www.wikidata.org/entity/Q2956264 
 Chantal Cadieux 
 http://www.wikidata.org/entity/P19 
 lieu de naissance Total: 230, Shown: 20

On trouve 230 références.

### D'autres villes

Refaisons l'exercice avec nos villes exemples pour comparer:

In [15]:
SELECT distinct ?writer ?writerLabel ?relation ?relationLabel ?city ?cityLabel
WHERE
{
  VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
  {?writer wdt:P106 wd:Q36180 .} UNION { ?writer wdt:P106 wd:Q482980 . } UNION { ?writer wdt:P106 wd:Q49757 . }
  ?writer ?rel ?city .
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
  ?relation wikibase:directClaim ?rel .
}

writer 
 writerLabel 
 relation 
 relationLabel 
 city 
 cityLabel http://www.wikidata.org/entity/Q231815 
 Pauley Perrette 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q298025 
 Joss Whedon 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q299194 
 Aaron Sorkin 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q310367 
 Jon Cryer 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q131240 
 Anna Politkovskaïa 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q179282 
 Steven Weinberg 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q240370 
 Robert Riskin 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q282517 
 Jean Liedloff 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q182870 
 Mario Puzo 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q139850 
 Zack O'Malley Greenburg 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q150651 
 George Carlin 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q284025 
 Carol Gilligan 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q165578 
 Jack Zipes 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q201418 
 Abigail Breslin 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q236950 
 Erica Jong 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q242643 
 Naomi Novik 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q208101 
 Joseph Heller 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q241299 
 Julia Ward Howe 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q287713 
 Ruth Chatterton 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York http://www.wikidata.org/entity/Q187765 
 Henry Miller 
 http://www.wikidata.org/entity/P19 
 lieu de naissance 
 http://www.wikidata.org/entity/Q60 
 New York Total: 16566, Shown: 20

_(on doit utiliser une UNION plutôt qu'une énumération VALUES dans la requête précédente pour éviter un "time out" de la requête...)_

Et si on veut un décompte du nombre d'auteurs ayant des relations avec les villes en question, par ville:

_(cette requête prend trop de temps à l'exécuter pour toutes les villes, on sépare en deux lots...)_

In [17]:
SELECT ?cityLabel (count(distinct ?writer) as ?writerCount)
WHERE
{
  VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 }
  {?writer wdt:P106 wd:Q36180 .} UNION { ?writer wdt:P106 wd:Q482980 . } UNION { ?writer wdt:P106 wd:Q49757 . }
  ?writer ?rel ?city
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
  ?relation wikibase:directClaim ?rel .
}
GROUP BY ?cityLabel

cityLabel 
 writerCount Barcelone 
 2000 Angoulême 
 31 Dublin 
 519 Québec 
 204 Total: 4, Shown: 4

In [25]:
SELECT ?cityLabel (count(distinct ?writer) as ?writerCount)
WHERE
{
  VALUES ?city { wd:Q490 wd:Q5083 }
  {?writer wdt:P106 wd:Q36180 .} UNION { ?writer wdt:P106 wd:Q482980 . } UNION { ?writer wdt:P106 wd:Q49757 . }
  ?writer ?rel ?city
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
  ?relation wikibase:directClaim ?rel .
}
GROUP BY ?cityLabel

cityLabel 
 writerCount Milan 
 1098 Seattle 
 238 Total: 2, Shown: 2

In [26]:
SELECT ?cityLabel (count(distinct ?writer) as ?writerCount)
WHERE
{
  VALUES ?city { wd:Q340 wd:Q90 wd:Q60 }
  {?writer wdt:P106 wd:Q36180 .} UNION { ?writer wdt:P106 wd:Q482980 . } UNION { ?writer wdt:P106 wd:Q49757 . }
  ?writer ?rel ?city
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
  ?relation wikibase:directClaim ?rel .
}
GROUP BY ?cityLabel

cityLabel 
 writerCount New York 
 3514 Paris 
 5855 Montréal 
 842 Total: 3, Shown: 3

### Types de relations (toutes villes)

Si on veut connaître les types de relations les plus fréquents entre les auteurs et les villes (autant pour Québec que pour les autres villes de notre échantillon).

In [16]:
SELECT ?relation (count(distinct ?writer) as ?writerCount)
WHERE
{
  VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
  {?writer wdt:P106 wd:Q36180 .} UNION { ?writer wdt:P106 wd:Q482980 . } UNION { ?writer wdt:P106 wd:Q49757 . }
  ?writer ?rel ?city
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
  ?relation wikibase:directClaim ?rel .
}
GROUP BY ?relation


relation 
 writerCount http://www.wikidata.org/entity/P108 
 2 http://www.wikidata.org/entity/P101 
 1 http://www.wikidata.org/entity/P19 
 7601 http://www.wikidata.org/entity/P69 
 1 http://www.wikidata.org/entity/P119 
 16 http://www.wikidata.org/entity/P551 
 931 http://www.wikidata.org/entity/P276 
 5 http://www.wikidata.org/entity/P740 
 7 http://www.wikidata.org/entity/P937 
 1271 http://www.wikidata.org/entity/P291 
 1 http://www.wikidata.org/entity/P840 
 1 http://www.wikidata.org/entity/P159 
 1 http://www.wikidata.org/entity/P20 
 6647 Total: 13, Shown: 13

Les relations les plus fréquentes sont donc le lieu de naissance (P19), de décès (P20), de résidence (P551) et de travail (P937).

La ville de scolarité semble très peu documentée, mais on peut émettre l'hypothèse que si on passe par l'institution d'enseignement, on aura plus de résultats. Par exemple, pour Québec:


In [19]:
SELECT distinct ?writer ?writerLabel ?school ?schoolLabel
WHERE
{
  {?writer wdt:P106 wd:Q36180 .} UNION { ?writer wdt:P106 wd:Q482980 . } UNION { ?writer wdt:P106 wd:Q49757 . }
  ?writer wdt:P69 ?school .
  ?school (wdt:P159*/wdt:P131*) wd:Q2145
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}


writer 
 writerLabel 
 school 
 schoolLabel http://www.wikidata.org/entity/Q3506003 
 Suzanne Martel 
 http://www.wikidata.org/entity/Q8077976 
 École des Ursulines, Quebec http://www.wikidata.org/entity/Q3165792 
 Jean-François Nadeau 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3292181 
 Marie-Paul Ross 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q19957875 
 Gilles Julien 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3171046 
 Jean Bédard 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q16663531 
 Martin Blais 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3507296 
 Sylvie Bérard 
 http://www.wikidata.org/entity/Q326282 
 Université du Québec http://www.wikidata.org/entity/Q15406975 
 Élizabeth Turgeon 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q1150878 
 Noël Simard 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3264759 
 Luc Bureau 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q55723035 
 Jean-Claude Castex 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3179881 
 Jocelyne Saucier 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q3310716 
 Michel Roberge 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q29919287 
 Sylvain Rivière 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q55670280 
 Maurice Hébert 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q580971 
 Louis-Honoré Fréchette 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q2371603 
 François Létourneau 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q2748972 
 Ferdinand Nahimana 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q16021010 
 André Jacques 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval http://www.wikidata.org/entity/Q2851659 
 Annie Groovie 
 http://www.wikidata.org/entity/Q1067935 
 Université Laval Total: 195, Shown: 20

## Exploration sur les livres

### De Québec

On veut voir les oeuvres écrites et les éditions (en utilisant donc le modèle à deux niveaux proposé par le projet Wikidata Livres) qui ont des liens avec Québec. D'abord pour les oeuvres:

In [20]:
SELECT distinct ?work ?workLabel
WHERE
{
    ?work wdt:P31/wdt:P279* wd:Q47461344 .
    ?work ?relation wd:Q2145 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}

Error: SPARQL error: EndPointInternalError: endpoint returned code 500 and response. 

Response:
b'SPARQL-QUERY: queryStr=SELECT distinct ?work ?workLabel\nWHERE\n{\n    ?work wdt:P31/wdt:P279* wd:Q47461344 .\n    ?work ?relation wd:Q2145 .\n    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }\n}\njava.util.concurrent.TimeoutException\n\tat java.util.concurrent.FutureTask.get(FutureTask.java:205)\n\tat com.bigdata.rdf.sail.webapp.BigdataServlet.submitApiTask(BigdataServlet.java:292)\n\tat com.bigdata.rdf.sail.webapp.QueryServlet.doSparqlQuery(QueryServlet.java:678)\n\tat com.bigdata.rdf.sail.webapp.QueryServlet.doGet(QueryServlet.java:290)\n\tat com.bigdata.rdf.sail.webapp.RESTServlet.doGet(RESTServlet.java:240)\n\tat com.bigdata.rdf.sail.webapp.MultiTenancyServlet.doGet(MultiTenancyServlet.java:273)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:687)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:790)\n\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)\n\tat org.wikidata.query.rdf.blazegraph.throttling.ThrottlingFilter.doFilter(ThrottlingFilter.java:320)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.throttling.SystemOverloadFilter.doFilter(SystemOverloadFilter.java:82)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat ch.qos.logback.classic.helpers.MDCInsertingServletFilter.doFilter(MDCInsertingServletFilter.java:49)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.QueryEventSenderFilter.doFilter(QueryEventSenderFilter.java:117)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.ClientIPFilter.doFilter(ClientIPFilter.java:43)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.RealAgentFilter.doFilter(RealAgentFilter.java:33)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634)\n\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)\n\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)\n\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)\n\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)\n\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1242)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)\n\tat org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)\n\tat org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\n\tat org.eclipse.jetty.server.Server.handle(Server.java:503)\n\tat org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364)\n\tat org.eclipse.jetty.server.Htt

L'exécution de la requête est trop longue. C'est la même chose si on limite à certains sous-types d'oeuvres écrites. C'est aussi la même chose si on veut simplement le décompte par type de relation:

In [27]:
SELECT ?relationLabel (count(distinct ?work) as ?workCount)
WHERE
{
    ?work ?rel wd:Q2145 .
    ?work wdt:P31/wdt:P279* wd:Q47461344 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
    ?relation wikibase:directClaim ?rel .
}
GROUP BY ?relationLabel

Error: SPARQL error: EndPointInternalError: endpoint returned code 500 and response. 

Response:
b'SPARQL-QUERY: queryStr=SELECT ?relationLabel (count(distinct ?work) as ?workCount)\nWHERE\n{\n    ?work ?rel wd:Q2145 .\n    ?work wdt:P31/wdt:P279* wd:Q47461344 .\n    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }\n    ?relation wikibase:directClaim ?rel .\n}\nGROUP BY ?relationLabel\njava.util.concurrent.TimeoutException\n\tat java.util.concurrent.FutureTask.get(FutureTask.java:205)\n\tat com.bigdata.rdf.sail.webapp.BigdataServlet.submitApiTask(BigdataServlet.java:292)\n\tat com.bigdata.rdf.sail.webapp.QueryServlet.doSparqlQuery(QueryServlet.java:678)\n\tat com.bigdata.rdf.sail.webapp.QueryServlet.doGet(QueryServlet.java:290)\n\tat com.bigdata.rdf.sail.webapp.RESTServlet.doGet(RESTServlet.java:240)\n\tat com.bigdata.rdf.sail.webapp.MultiTenancyServlet.doGet(MultiTenancyServlet.java:273)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:687)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:790)\n\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)\n\tat org.wikidata.query.rdf.blazegraph.throttling.ThrottlingFilter.doFilter(ThrottlingFilter.java:320)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.throttling.SystemOverloadFilter.doFilter(SystemOverloadFilter.java:82)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat ch.qos.logback.classic.helpers.MDCInsertingServletFilter.doFilter(MDCInsertingServletFilter.java:49)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.QueryEventSenderFilter.doFilter(QueryEventSenderFilter.java:117)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.ClientIPFilter.doFilter(ClientIPFilter.java:43)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.RealAgentFilter.doFilter(RealAgentFilter.java:33)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634)\n\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)\n\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)\n\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)\n\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)\n\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1242)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)\n\tat org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)\n\tat org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\n\tat org.eclipse.jetty.server.Server.handle(Server.java:503)\n\tat org.eclips

On va donc plutôt utiliser une requête sur des relations spécifiques liées à l'emplacement, en cherchant les oeuvres dont le sujet principal (P921), l'action (P840), ou ce qui est décrit (P180) est Québec:

In [23]:
SELECT distinct ?work ?workLabel
WHERE
{
    ?work (wdt:P31/wdt:P279*) wd:Q47461344 .
    ?work (wdt:P921|wdt:P840|wdt:P180) wd:Q2145 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}

work 
 workLabel http://www.wikidata.org/entity/Q886446 
 Spider Bones http://www.wikidata.org/entity/Q30602331 
 Bury Your Dead http://www.wikidata.org/entity/Q106893413 
 Réseau express de la Capitale - Le gouvernement dévoile sa vision du vaste réseau intégré de transport collectif dans la région métropolitaine de Québec http://www.wikidata.org/entity/Q106909351 
 Une pétition contre le troisième lien Québec-Lévis est lancée http://www.wikidata.org/entity/Q63648939 
 Joseph-Ernest Grégoire, quatre années de vie politique http://www.wikidata.org/entity/Q106571862 
 Ma ville http://www.wikidata.org/entity/Q97303233 
 View of Quebec from the Basin http://www.wikidata.org/entity/Q24439187 
 Q24439187 http://www.wikidata.org/entity/Q25043072 
 Q25043072 http://www.wikidata.org/entity/Q29042934 
 À fond de train Total: 10, Shown: 10

On peut également reprendre la requête originale, trop longue à exécuter, mais en se limitant aux oeuvres littéraires:

In [29]:
SELECT distinct ?work ?workLabel ?relation ?relationLabel
WHERE
{
    ?work wdt:P31/wdt:P279* wd:Q7725634 .
    ?work ?rel wd:Q2145 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
    ?relation wikibase:directClaim ?rel .
}

work 
 workLabel 
 relation 
 relationLabel http://www.wikidata.org/entity/Q886446 
 Spider Bones 
 http://www.wikidata.org/entity/P840 
 lieu de l'action http://www.wikidata.org/entity/Q106582950 
 Veillée 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q30310263 
 Les Révélations du crime ou Cambray et ses complices 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q51513428 
 Les oiseaux de la province de Québec 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q30602331 
 Bury Your Dead 
 http://www.wikidata.org/entity/P840 
 lieu de l'action http://www.wikidata.org/entity/Q51468513 
 L'élevage du cheval en Canada 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q22811578 
 Édits, ordonnances royaux, déclarations et arrêts du Conseil d'État du Roi concernant le Canada 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q51443292 
 Les arbres de commerce de la province de Québec 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q51515313 
 Ornithologie du Canada d'après la nomenclature de Baird 
 http://www.wikidata.org/entity/P291 
 lieu de publication Total: 9, Shown: 9

Ensuite pour les éditions, on cherche toutes celles ayant un lien avec Québec, peu importe le lien:

In [24]:
SELECT distinct ?edition ?editionLabel ?relation ?relationLabel
WHERE
{
    ?edition (wdt:P31/wdt:P279*) wd:Q3331189 .
    ?edition ?rel wd:Q2145 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
    ?relation wikibase:directClaim ?rel .
}

edition 
 editionLabel 
 relation 
 relationLabel http://www.wikidata.org/entity/Q55238826 
 La tempérance et les destinées du Canada 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q101406016 
 Mena’sen 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q50828733 
 Trois contes sauvages 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q52685087 
 Québec en 1900 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q26037561 
 Le Créole patriote (1792-1794) : un pont entre deux Révolutions 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q54733279 
 Portraits et Pastels Littéraires 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q107228267 
 Les frères Sisters 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q51515316 
 Ornithologie du Canada : quelques groupes d'après la nomenclature du Smithsonian Institution, de Washington 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q51517160 
 Les pêcheries du Canada 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q78880939 
 Les jésuites et l'esclavage 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q48544534 
 Les comtés de Rimouski, de Matane et de Témiscouata 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q51502553 
 Les mauvaises herbes dans la province de Québec et différents moyens de les détruire 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q92078844 
 Quelques notes sur Antoine de Lamothe de Cadillac 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q26205278 
 Nouveau Chemin de la croix 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q63615887 
 Dialogue avec l'histoire : les collections de la Bibliothèque de l'Université Laval 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q97303233 
 View of Quebec from the Basin 
 http://www.wikidata.org/entity/P180 
 dépeint http://www.wikidata.org/entity/Q51513428 
 Les oiseaux de la province de Québec 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q28976060 
 Le chemin de fer du lac Saint-Jean 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q60647567 
 Ce que disait la flamme… 
 http://www.wikidata.org/entity/P291 
 lieu de publication http://www.wikidata.org/entity/Q61864645 
 Mount Hermon Cemetery : rules and regulations. 
 http://www.wikidata.org/entity/P291 
 lieu de publication Total: 35, Shown: 20

On note que certains éléments reviennent dans les deux listes précédentes: ils ont été documentés à la fois comme livre et comme édition.

Même si cela ne respecte pas les conventions proposées par le projet Wikidata Livres, on veut voir si des entités sont classées comme livres (et pas oeuvre écrite ni édition) et sont liées à Québec:

In [16]:
SELECT distinct ?book ?bookLabel
WHERE
{
    ?book (wdt:P31/wdt:P279*) wd:Q571 .
    ?book ?relation wd:Q2145 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}

book 
 bookLabel http://www.wikidata.org/entity/Q98073693 
 Le Palais Montcalm : 75 ans d'émotions http://www.wikidata.org/entity/Q51512145 
 Nouveau manuel complet d'industrie laitière pour la province de Québec http://www.wikidata.org/entity/Q61864645 
 Mount Hermon Cemetery : rules and regulations. http://www.wikidata.org/entity/Q51513444 
 Les oisseaux insectivores Total: 4, Shown: 4

### Ville de publication et comparaison avec d'autres villes

Pour continuer à avoir des points de repère, on voudra comparer le nombre d'oeuvres *publiées* à Québec, comparées à celles publiées dans les autres villes de notre échantillon. Première requête pour voir quelques exemples (en se limitant aux oeuvres littéraires, et en segmentant les villes en deux groupes, pour les raisons de performance déjà évoquées):

In [30]:
SELECT ?cityLabel (count(distinct ?work) as ?workCount)
WHERE
{
    VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 }
    ?work wdt:P31/wdt:P279* wd:Q7725634 .
    ?work wdt:P291 ?city .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}
GROUP BY ?cityLabel

cityLabel 
 workCount Barcelone 
 119 Dublin 
 24 Québec 
 7 Angoulême 
 1 Total: 4, Shown: 4

In [31]:
SELECT ?cityLabel (count(distinct ?work) as ?workCount)
WHERE
{
    VALUES ?city { wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
    ?work wdt:P31/wdt:P279* wd:Q7725634 .
    ?work wdt:P291 ?city .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}
GROUP BY ?cityLabel

cityLabel 
 workCount Paris 
 1507 New York 
 212 Montréal 
 12 Milan 
 68 Seattle 
 2 Total: 5, Shown: 5

Et si on refait le même exercice sur la base des éditions:

In [34]:
SELECT ?cityLabel (count(distinct ?edition) as ?editionCount)
WHERE
{
    VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
    ?edition wdt:P31/wdt:P279* wd:Q3331189 .
    ?edition wdt:P291 ?city .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}
GROUP BY ?cityLabel

cityLabel 
 editionCount New York 
 1362 Paris 
 3278 Milan 
 195 Montréal 
 102 Seattle 
 44 Barcelone 
 1554 Dublin 
 64 Québec 
 34 Total: 8, Shown: 8

Pour compléter le portrait, nous pouvons faire le décompte des oeuvres qui traitent, ou dont l'action se déroule, dans les villes concernées.

In [3]:
SELECT ?cityLabel (count(distinct ?work) as ?workCount)
WHERE
{
    VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 }
    ?work (wdt:P921|wdt:P840|wdt:P180) ?city .
    ?work wdt:P31/wdt:P279* wd:Q7725634 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}
GROUP BY ?cityLabel

cityLabel 
 workCount Dublin 
 38 Québec 
 2 Barcelone 
 28 Total: 3, Shown: 3

In [4]:
SELECT ?cityLabel (count(distinct ?work) as ?workCount)
WHERE
{
    VALUES ?city { wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
    ?work (wdt:P921|wdt:P840|wdt:P180) ?city .
    ?work wdt:P31/wdt:P279* wd:Q7725634 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}
GROUP BY ?cityLabel

Error: SPARQL error: EndPointInternalError: endpoint returned code 500 and response. 

Response:
b'SPARQL-QUERY: queryStr=SELECT ?cityLabel (count(distinct ?work) as ?workCount)\nWHERE\n{\n    VALUES ?city { wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }\n    ?work (wdt:P921|wdt:P840|wdt:P180) ?city .\n    ?work wdt:P31/wdt:P279* wd:Q7725634 .\n    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }\n}\nGROUP BY ?cityLabel\njava.util.concurrent.TimeoutException\n\tat java.util.concurrent.FutureTask.get(FutureTask.java:205)\n\tat com.bigdata.rdf.sail.webapp.BigdataServlet.submitApiTask(BigdataServlet.java:292)\n\tat com.bigdata.rdf.sail.webapp.QueryServlet.doSparqlQuery(QueryServlet.java:678)\n\tat com.bigdata.rdf.sail.webapp.QueryServlet.doGet(QueryServlet.java:290)\n\tat com.bigdata.rdf.sail.webapp.RESTServlet.doGet(RESTServlet.java:240)\n\tat com.bigdata.rdf.sail.webapp.MultiTenancyServlet.doGet(MultiTenancyServlet.java:273)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:687)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:790)\n\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)\n\tat org.wikidata.query.rdf.blazegraph.throttling.ThrottlingFilter.doFilter(ThrottlingFilter.java:320)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.throttling.SystemOverloadFilter.doFilter(SystemOverloadFilter.java:82)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat ch.qos.logback.classic.helpers.MDCInsertingServletFilter.doFilter(MDCInsertingServletFilter.java:49)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.QueryEventSenderFilter.doFilter(QueryEventSenderFilter.java:117)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.ClientIPFilter.doFilter(ClientIPFilter.java:43)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.RealAgentFilter.doFilter(RealAgentFilter.java:33)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634)\n\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)\n\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)\n\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)\n\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)\n\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1242)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)\n\tat org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)\n\tat org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\n\tat org.eclipse.jetty.server.Server.handle(Ser

Et enfin, même question, pour les éditions.

In [ ]:
SELECT ?cityLabel (count(distinct ?edition) as ?editionCount)
WHERE
{
    VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
    ?edition (wdt:P921|wdt:P840|wdt:P180) ?city .
    ?edition wdt:P31/wdt:P279* wd:Q3331189 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}
GROUP BY ?cityLabel

## À travers les maisons d'édition

Si on émet l'hypothèse que les villes de de publication ne sont pas bien documentées dans les oeuvres ou les éditions, on peut reprendre l'analyse en s'appuyant sur le lieux associé à la maison d'édition.

Par exemple, on peut lister les éditions publiées par des maisons de Québec:

In [33]:
SELECT ?edition ?editionLabel (GROUP_CONCAT(DISTINCT ?editeur; separator = ", ") AS ?editeurs)
WHERE
{
    ?edition     wdt:P31/wdt:P279*                       wd:Q3331189   .
    ?edition     wdt:P123                                ?editeur     .
    ?editeur     (wdt:P159|wdt:P131)/wdt:P131*           wd:Q2145      .
  
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}
GROUP BY ?edition ?editionLabel

edition 
 editionLabel 
 editeurs http://www.wikidata.org/entity/Q77009625 
 Dictionnaire des artistes de langue française en Amérique du Nord 
 http://www.wikidata.org/entity/Q2338135, http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q63615887 
 Dialogue avec l'histoire : les collections de la Bibliothèque de l'Université Laval 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q63373321 
 Bibliothèque de l'Université Laval : 165 ans d'histoire : 1852-2017 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q58341258 
 Statistiques sociales 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q52182553 
 Le Journalisme à l'heure du Net 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q98073693 
 Le Palais Montcalm : 75 ans d'émotions 
 http://www.wikidata.org/entity/Q98073919, http://www.wikidata.org/entity/Q2986483 http://www.wikidata.org/entity/Q107228267 
 Les frères Sisters 
 http://www.wikidata.org/entity/Q16684371 http://www.wikidata.org/entity/Q91949031 
 Frontenac et ses amis 
 http://www.wikidata.org/entity/Q40882270 http://www.wikidata.org/entity/Q78880939 
 Les jésuites et l'esclavage 
 http://www.wikidata.org/entity/Q3402573 http://www.wikidata.org/entity/Q56816764 
 Le registre de Sillery (1638-1690) 
 http://www.wikidata.org/entity/Q3402573 http://www.wikidata.org/entity/Q54165640 
 Ashton Leblond, juste du vrai! : un rêve réalisé... une frite à la fois! 
 http://www.wikidata.org/entity/Q3579394 http://www.wikidata.org/entity/Q40881798 
 Allumez vos lampes, s’il vous plaît !!! 
 http://www.wikidata.org/entity/Q40882270 http://www.wikidata.org/entity/Q51513444 
 Les oisseaux insectivores 
 http://www.wikidata.org/entity/Q40882270 http://www.wikidata.org/entity/Q51513428 
 Les oiseaux de la province de Québec 
 http://www.wikidata.org/entity/Q40882270 http://www.wikidata.org/entity/Q51502553 
 Les mauvaises herbes dans la province de Québec et différents moyens de les détruire 
 http://www.wikidata.org/entity/Q40882270 http://www.wikidata.org/entity/Q51468513 
 L'élevage du cheval en Canada 
 http://www.wikidata.org/entity/Q40882270 http://www.wikidata.org/entity/Q29018702 
 Le Traité de Paris de 1763 en bref 
 http://www.wikidata.org/entity/Q3579664 Total: 17, Shown: 17

Puis on fait la même chose pour les oeuvres:

In [36]:
SELECT ?oeuvre ?oeuvreLabel (GROUP_CONCAT(DISTINCT ?editeur; separator = ", ") AS ?editeurs)
WHERE
{
    ?oeuvre     wdt:P31/wdt:P279*                       wd:Q7725634   .
    ?oeuvre     wdt:P123                                ?editeur      .
    ?editeur     (wdt:P159|wdt:P131)/wdt:P131*          wd:Q2145      .
  
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}
GROUP BY ?oeuvre ?oeuvreLabel

oeuvre 
 oeuvreLabel 
 editeurs http://www.wikidata.org/entity/Q63872069 
 Aides-soignantes et autres funambules du soin : entre nécessités et arts de faire 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q63863366 
 Jeunes et djihadisme : les conversions interdites 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q63863193 
 Peinture et société au Québec 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q63863087 
 Jean-Charles-Bonenfant et l'esprit des institutions 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q63863048 
 Pour une reconnaissance de la laïcité au Québec : enjeux philosophiques, politiques et juridiques 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q63862938 
 Penser la laïcité québécoise : fondements et défense d'une laïcité ouverte au Québec 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q63862850 
 Les catholiques québécois et la laïcité 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q63862605 
 Laïcité et signes religieux à l'école 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q63862229 
 L'histoire des neurosciences à Québec : cinquante ans de recherche sur le cerveau et ses maladies 
 http://www.wikidata.org/entity/Q3402572 http://www.wikidata.org/entity/Q18354521 
 Sanaaq 
 http://www.wikidata.org/entity/Q1067935 http://www.wikidata.org/entity/Q104394234 
 Montréal et la bombe 
 http://www.wikidata.org/entity/Q3579664 http://www.wikidata.org/entity/Q67086226 
 Psychologie cognitive : une approche de traitement de l'information 
 http://www.wikidata.org/entity/Q3402573 http://www.wikidata.org/entity/Q63861521 
 L'apparition du Nord : selon Gérard Mercator 
 http://www.wikidata.org/entity/Q3579664 http://www.wikidata.org/entity/Q60607969 
 La Médecine en Nouvelle-France: Les chirurgiens de Montréal, 1642-1760 
 http://www.wikidata.org/entity/Q3579664 http://www.wikidata.org/entity/Q51513428 
 Les oiseaux de la province de Québec 
 http://www.wikidata.org/entity/Q40882270 http://www.wikidata.org/entity/Q51468513 
 L'élevage du cheval en Canada 
 http://www.wikidata.org/entity/Q40882270 http://www.wikidata.org/entity/Q51443292 
 Les arbres de commerce de la province de Québec 
 http://www.wikidata.org/entity/Q40882270 Total: 17, Shown: 17

On peut répéter l'exercice et noter le nombre d'éditions associées aux villes de notre échantillon à travers nos éditeurs:

In [37]:
SELECT ?cityLabel (count(distinct ?edition) as ?editionCount)
WHERE
{
    VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
    ?edition     wdt:P31/wdt:P279*                       wd:Q3331189   .
    ?edition     wdt:P123                                ?editeur      .
    ?editeur     (wdt:P159|wdt:P131)/wdt:P131*           ?city         .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}
GROUP BY ?cityLabel

cityLabel 
 editionCount Paris 
 2529 New York 
 1850 Barcelone 
 149 Québec 
 17 Milan 
 1026 Dublin 
 25 Seattle 
 21 Montréal 
 43 Total: 8, Shown: 8

Et le faire sur le nombre d'oeures associées aux villes de notre échantillon à travers nos éditeurs:

In [38]:
SELECT ?cityLabel (count(distinct ?oeuvre) as ?oeuvreCount)
WHERE
{
    VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }
    ?oeuvre     wdt:P31/wdt:P279*                       wd:Q7725634   .
    ?oeuvre     wdt:P123                                ?editeur      .
    ?editeur     (wdt:P159|wdt:P131)/wdt:P131*          ?city         .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}
GROUP BY ?cityLabel

Error: SPARQL error: EndPointInternalError: endpoint returned code 500 and response. 

Response:
b'SPARQL-QUERY: queryStr=SELECT ?cityLabel (count(distinct ?oeuvre) as ?oeuvreCount)\nWHERE\n{\n    VALUES ?city { wd:Q2145 wd:Q134346 wd:Q1492 wd:Q1761 wd:Q490 wd:Q5083 wd:Q340 wd:Q90 wd:Q60 }\n    ?oeuvre     wdt:P31/wdt:P279*                       wd:Q7725634   .\n    ?oeuvre     wdt:P123                                ?editeur      .\n    ?editeur     (wdt:P159|wdt:P131)/wdt:P131*          ?city         .\n    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }\n}\nGROUP BY ?cityLabel\njava.util.concurrent.TimeoutException\n\tat java.util.concurrent.FutureTask.get(FutureTask.java:205)\n\tat com.bigdata.rdf.sail.webapp.BigdataServlet.submitApiTask(BigdataServlet.java:292)\n\tat com.bigdata.rdf.sail.webapp.QueryServlet.doSparqlQuery(QueryServlet.java:678)\n\tat com.bigdata.rdf.sail.webapp.QueryServlet.doGet(QueryServlet.java:290)\n\tat com.bigdata.rdf.sail.webapp.RESTServlet.doGet(RESTServlet.java:240)\n\tat com.bigdata.rdf.sail.webapp.MultiTenancyServlet.doGet(MultiTenancyServlet.java:273)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:687)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:790)\n\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:865)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)\n\tat org.wikidata.query.rdf.blazegraph.throttling.ThrottlingFilter.doFilter(ThrottlingFilter.java:320)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.throttling.SystemOverloadFilter.doFilter(SystemOverloadFilter.java:82)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat ch.qos.logback.classic.helpers.MDCInsertingServletFilter.doFilter(MDCInsertingServletFilter.java:49)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.QueryEventSenderFilter.doFilter(QueryEventSenderFilter.java:117)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.ClientIPFilter.doFilter(ClientIPFilter.java:43)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)\n\tat org.wikidata.query.rdf.blazegraph.filters.RealAgentFilter.doFilter(RealAgentFilter.java:33)\n\tat org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1634)\n\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)\n\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)\n\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)\n\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)\n\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)\n\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1242)\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)\n\tat org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)\n\tat org.eclipse.jetty.server.handler.HandlerCollection

## Vrac / à explorer

Je vois Nicolas Dickner dans les auteurs qui ont étudié à Québec, à l'Université Laval. Sa fiche contient des oeuvres notables, Tarmac et Nikolski. Les deux fiches portent sur les oeuvres et ne contiennent pas de référence aux éditions (et donc pas aux ISBN). Aucune fiche ne fait de lien avec les Éditions Alto.

## Autres notes / réflexions

### Les sous-classes des oeuvres écrites

La classe des oeuvres écrites (Q47461344) comporte un très grand nombre (plus de 70 000) de sous-classes, ce qui rend la plupart des requêtes concernant des entités de cette classe ou de ses sous-classes impraticables (elles génèrent des dépassement du délai d'exécution permis). La stratégie que nous utiliserons sera de s'intéresser seulement à certaines sous-classes de premier niveau, soit:

In [8]:
SELECT distinct ?subclass ?subclassLabel
WHERE
{
    ?subclass wdt:P279 wd:Q47461344 .
    SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],fr,en". }
}

subclass 
 subclassLabel http://www.wikidata.org/entity/Q191067 
 article http://www.wikidata.org/entity/Q234460 
 texte http://www.wikidata.org/entity/Q277759 
 série de livres http://www.wikidata.org/entity/Q1433985 
 broadside http://www.wikidata.org/entity/Q1385450 
 thèse ou écrit académique http://www.wikidata.org/entity/Q1501855 
 programme électoral http://www.wikidata.org/entity/Q3092752 
 Apocryphe http://www.wikidata.org/entity/Q3030630 
 dissertation http://www.wikidata.org/entity/Q3099732 
 rapport technique http://www.wikidata.org/entity/Q7725634 
 œuvre littéraire http://www.wikidata.org/entity/Q1376568 
 appendix http://www.wikidata.org/entity/Q1640824 
 inscription lapidaire http://www.wikidata.org/entity/Q2352616 
 catalogue http://www.wikidata.org/entity/Q15087423 
 norme ISO http://www.wikidata.org/entity/Q94709889 
 Catholic church records http://www.wikidata.org/entity/Q3749265 
 fragment http://www.wikidata.org/entity/Q16324495 
 Compilation http://www.wikidata.org/entity/Q12106333 
 recueil de poèmes http://www.wikidata.org/entity/Q1440451 
 fragment collection http://www.wikidata.org/entity/Q1440453 
 fragment Total: 39, Shown: 20

Nous avons retenu les sous-classes suivantes, liées d'une certaine façon au monde du livre:

* série de livres  wd:Q277759
* thèse ou écrit académique wd:Q1385450
* apocryphe wd:Q3092752
* dissertation wd:Q3030630
* rapport technique wd:Q3099732
* catalogue wd:Q2352616
* compilation wd:Q16324495
* recueil de poèmes wd:Q12106333
* fragment wd:Q1440453 («twittérature»)
* œuvre littéraire wd:Q7725634
* œuvre scientifique (Q11826511)
* œuvre collective (Q3594128
* manga (Q8274)
* bande dessinée (Q1004)
* dictionnaire (Q23622)
* ouvrage de référence (Q13136)
* œuvre académique (Q55915575)
* front and back matter of a book(Q61782498)
* Guide d'achat du vin (Q63994109)

Tout ça semble un peu arbitraire ou étrange (pourquoi les guides d'achat de vin sont une sous-classe directe des oeuvres écrites, mais pas les autres guides d'achat? manga est à la fois une sous-classe d'oeuvre écrite et de bande dessinée, est-ce nécessaire de le conserver dans notre sélection?).

On pourrait donc utiliser `VALUES ?workType { wd:Q277759 wd:Q1385450 wd:Q3092752 wd:Q3030630 wd:Q3099732 wd:Q2352616 wd:Q16324495 wd:Q12106333 wd:Q1440453 wd:Q7725634 wd:Q11826511 wd:Q3594128 wd:Q8274 wd:Q1004 wd:Q23622 wd:Q13136 wd:Q55915575 wd:Q61782498 wd:Q63994109 }` dans nos requêtes.

Toutefois, après quelques essais, ça ne règle pas vraiment le problème du temps d'exécution.

# Si on devait résumer la démarche...

C'est très difficile d'obtenir des données de Wikidata sur les livres.

Cela s'explique de différentes raisons:

* La complexité du modèle théorique, qui demande de distinguer oeuvre et éditions, qui n'est pas toujours respecté
* Le très grand nombre de sous-classes de la classe des oeuvres écrites, qui rend toute requête sur cette classe et ses sous-classes très longue (menant souvent à un "time out")

Cela s'explique également par la nature des données qu'on interroge, qui ne sont pas toutes complètes (des livres qui ne sont pas liés à leur maison d'édition ou à leur lieux de publication).

Notons de plus que nos requêtes ne sont probablement pas toutes optimales, à la fois sur le délai d'exécution, que sur la façon d'interrog

Néanmoins, sur l'ensemble des indicateurs pour lesquels on arrive à avoir des résultats et des comparaisons avec les autres villes de notre échantillon (le nombre d'éditeurs situés dans la ville, le nombre d'oeuvres ou d'éditions qui y sont publiées, le nombre d'oeuvres ou d'éditions publiées par des éditeurs situés dans la ville), Québec ne fait pas particulièrement bonne figure (sans non plus être la dernière).